File: app/src/main/java/com/nextcloud/client/preferences/PreferencesModule.java
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2019 Chris Narkiewicz <hello@ezaquarii.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.nextcloud.client.preferences;

import android.content.Context;
import android.content.SharedPreferences;

import com.nextcloud.client.account.UserAccountManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PreferencesModule {

    @Provides
    @Singleton
    public SharedPreferences sharedPreferences(Context context) {
        return android.preference.PreferenceManager.getDefaultSharedPreferences(context);
    }

    @Provides
    @Singleton
    public AppPreferences appPreferences(Context context,
                                         SharedPreferences sharedPreferences,
                                         UserAccountManager userAccountManager) {
        return new AppPreferencesImpl(context, sharedPreferences, userAccountManager);
    }
}

File: app/src/main/java/com/nextcloud/client/preferences/PreferencesModule.kt
package com.nextcloud.client.preferences

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import com.nextcloud.client.account.UserAccountManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PreferencesModule {
    @Provides
    @Singleton
    fun sharedPreferences(context: Context?): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    @Provides
    @Singleton
    fun appPreferences(
        context: Context,
        sharedPreferences: SharedPreferences,
        userAccountManager: UserAccountManager
    ): AppPreferences {
        return AppPreferencesImpl(context, sharedPreferences, userAccountManager)
    }
}